### PR TITLE
Add optional hooks to lint and format

### PR DESCRIPTION
**Pull request by zacanger**
Adds optional hooks using husky, lint-staged, and opt. Also adds a HACKING.md with info on the hooks, and moves "how to get up and running" bit from the registry readme to HACKING.md.

I added these because I keep forgetting to run Prettier and the linters manually, and figure other folks might, too.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Entropic: a federated package registry for anything
-[![All Contributors](https://img.shields.io/badge/all_contributors-15-orange.svg?style=flat-square)](#contributors)
+
```
Comment (zacanger) [inline README.md]: Had to make some adjustments to this file because of the linting hook.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
+[![All Contributors](https://img.shields.io/badge/all_contributors-15-orange.svg?style=flat-square)](#contributors) [![CircleCI](https://circleci.com/gh/entropic-dev/entropic.svg?style=svg)](https://circleci.com/gh/entropic-dev/entropic)
```
Comment (zacanger) [inline README.md]: Added a Circle badge here

File: cli/lib/fetch.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-"use strict";
+'use strict';
```
Comment (zacanger) [inline cli/lib/fetch.js]: Prettier

Comment (olingern) [reply cli/lib/fetch.js]: I think #183 takes care of this, but hopefully #149 can prevent in the future :slightly_smiling_face:

**Comment by zacanger** [comment]
Thanks for merging! This will save me so many brain cycles each time I commit.

Also it's totally opt-in, so if anyone comes across this PR, don't feel pressured to use thee hooks. It's just a nice convenience if you're used to some other coding style.

**Pull request merged**